src/utils/queryBuilder.ts

```diff
@@ -210,22 +216,36 @@ const _generateDateRangeQuery = (
   } else if (fields?.date?.fdy && fields?.date?.tdy) {
     filterBlock.push(..._generateRangeBlock(fields.date));
   }
+  if (fields?.classify?.level && fields.classify.parent) {
+    filterBlock.push(_generateTermsBlock(level && levelMap[level], parentArray));
+  }
   return filterBlock;
 };
 
 const generateSearchQuery = (searchBuilderPayload: ISearchBuilderPayload): estypes.SearchRequest => {
   const queryPayload: estypes.SearchRequest = _generateQuery(searchBuilderPayload);
   const { searchFieldsObject, docId = '' } = searchBuilderPayload;
   const { filters } = (searchFieldsObject as ISearchPayload) ?? {};
+  const { level, parent } = (searchFieldsObject?.fields.classify as ISearchPayload) ?? {};
+  const levelMap = {
+    1: 'OrgNceaClassifiers.code.keyword',
+    2: 'OrgNceaClassifiers.classifiers.code.keyword',
+    3: 'OrgNceaClassifiers.classifiers.classifiers.code.keyword',
+  };
```

Comment:
move this const levelMap to a common place and use it. code duplicate

Reply:
Will address this in next pr